src/jaxsim/rbda/contacts/common.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import abc
+from typing import Any
+
+import jaxsim.terrain
+import jaxsim.typing as jtp
+
+
+class ContactsState(abc.ABC):
+    """
+    Abstract class storing the state of the contacts model.
+    """
+
+    @classmethod
+    @abc.abstractmethod
+    def build(cls, **kwargs) -> ContactsState:
+        """
+        Build the contact state object.
+
+        Returns:
+            The contact state object.
+        """
+
+    @classmethod
+    @abc.abstractmethod
+    def zero(cls, **kwargs) -> ContactsState:
+        """
+        Build a zero contact state.
+
+        Returns:
+            The zero contact state.
+        """
+
+    @abc.abstractmethod
+    def valid(self, **kwargs) -> bool:
+        """
+        Check if the contacts state is valid.
+        """
+
+
+class ContactsParams(abc.ABC):
+    """
+    Abstract class representing the parameters of a contact model.
+    """
+
+    @classmethod
+    @abc.abstractmethod
+    def build(cls) -> ContactsParams:
+        """
+        Create a `ContactsParams` instance with specified parameters.
+        Returns:
+            The `ContactsParams` instance.
+        """
+
```

Comment:
```suggestion
        """
        pass
```

Reply:
It was to avoid [`unnecessary-pass / W0107`](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/unnecessary-pass.html)

Reply:
Mmh personal taste, I don't like it. That works only because there is a docstring. If you don't like the pass, use `...`. This code is never executed anyway, there's no performance to be optimized, and I like explicitly seeing a no-op body function.